plutus-core/untyped-plutus-core/src/UntypedPlutusCore/Evaluation/Machine/Cek/Internal.hs

```diff
@@ -648,6 +650,20 @@ evalBuiltinApp fun term runtime = case runtime of
     _ -> pure $ VBuiltin fun term runtime
 {-# INLINE evalBuiltinApp #-}
 
+data ArgsR uni fun ann
+    = NilArgsR
+    | ConsArgsR !(CekValue uni fun ann) !(ArgsR uni fun ann)
+    deriving stock (Show)
+
+toListArgsR :: ArgsR uni fun ann -> [CekValue uni fun ann]
+toListArgsR = go [] where
+    go acc NilArgsR             = acc
+    go acc (ConsArgsR arg args) = go (arg : acc) args
+
+appendArgsR :: ArgsR uni fun ann -> Context uni fun ann -> Context uni fun ann
```

Comment:
I found the name of this a little confusing. Not sure what else to call it.

Reply:
Well, I did manage to resist the temptation to call it [`chips`](https://github.com/pigworker/SSGEP-DataData/blob/26bc46552f188765d3e98a49270027390155e46c/STLC.lagda#L265-L273), don't make me question my hard-to-make decisions.